src/lib.rs

```diff
@@ -270,20 +273,23 @@ enum IpfsEvent {
 /// Configured Ipfs instace or value which can be only initialized.
 pub struct UninitializedIpfs<Types: IpfsTypes> {
     repo: Option<Repo<Types>>,
+    span: Span,
     keys: Keypair,
     options: IpfsOptions<Types>,
     moved_on_init: Option<Receiver<RepoEvent>>,
 }
 
 impl<Types: IpfsTypes> UninitializedIpfs<Types> {
-    /// Configures a new UninitializedIpfs with from the given options.
-    pub async fn new(options: IpfsOptions<Types>) -> Self {
+    /// Configures a new UninitializedIpfs with from the given options and optionally a span.
```

Comment:
```suggestion
    /// Configures a new UninitializedIpfs with from the given options and optionally a span.
    /// When given the `Span` all operations called through the created `Ipfs` type will be
    /// attached that span, along with all operations done in the background task as well as
    /// the tasks spawned by the underlying `libp2p::Swarm`. When left None, the tracing
    /// will still be enabled, but the events from the single `Ipfs` and it's underlying tasks will
    /// not be correlatable by a span.
```

Does make might though that this could be an builder method, which would need to be invoked before `start`? Then again, as this already needs to instrument a few futures this might be the only way to go.

Reply:
```suggestion
    /// Configures a new UninitializedIpfs with from the given options and optionally a span.
    /// If the span is not given, it is defaulted to `tracing::trace_span!("ipfs")`.
    ///
    /// The span is attached to all operations called on the later created `Ipfs` along with all
    /// operations done in the background task as well as tasks spawned by the underlying
    /// `libp2p::Swarm`.
```